api/app/models/offer.rb

```diff
@@ -1,9 +1,81 @@
 # frozen_string_literal: true
 
 # A class representing an offer. This class belongs to assignment, applicant and position.
+require 'json'
 class Offer < ApplicationRecord
   	belongs_to :assignment
-end
+
+
+    def get_deadline
+      offer = self.as_json
+      if offer[:send_date]
+        DateTime.parse(offer[:send_date]).days_ago(-21)
+      end
+    end
+
+    def format
+      offer = self.as_json
+
+      #These will be removed once I figure out the rate.
+      assignment = Assignment.find(offer["assignment_id"]).as_json
+      position = Position.find(assignment["position_id"]).as_json
+      applicant = Applicant.find(assignment["applicant_id"])
+      session = Session.find(position["session_id"])
+      if offer[:link]
+        offer[:link]= "#{ENV["domain"]}#{offer[:link]}"
+      end
+      data = {
+        position: offer["position_code"],
+        start_date: offer["position_start_date" ],
+        end_date: offer["position_end_date"],
+        hours: offer["pay_period_desc"],
+        first_name: offer["first_name"],
+        last_name: offer["last_name"],
+        session: session,
+        session_info: session.format,
+        deadline: self.get_deadline,
+      }
+      if offer[:send_date]
+        data[:deadline] = self.get_deadline
+      end
+      # the Liquid templating engine assumes strings instead of symbols
+      return offer.merge(data).with_indifferent_access
+    end
+
+    def instructor_format
+      offer = self.as_json
+
+      assignment = Assignment.find(offer["assignment_id"]).as_json
+      position = Position.find(assignment["position_id"]).as_json
+      applicant = Applicant.find(assignment["applicant_id"])
+
+      data = {
+        position: position["position_title"],
+        applicant: applicant.format,
+      }
+      excludes = [
+        :accept_date,
+        :commentary,
+        :hr_status,
+        :link,
+        :nag_count,
+        :print_time,
+        :send_date,
+        :signature,
+        :year,
+        :session,
+      ]
+      # the Liquid templating engine assumes strings instead of symbols
+      return offer.merge(data).except(*excludes).with_indifferent_access
+    end
+
+    private
+    def randomize_id
```

Comment:
Why are we randomizing the id?  The id is a field set by rails, I don't think we should touch this.

Reply:
We randomize the ID so it is not guessable. We don't want the rails default